changelog/2604.doc

```diff
@@ -0,0 +1 @@
+In examples on working with custom markers, add examples demonstrating the usage of `pytest.mark.MARKER_NAME.with_args` in comparison with `pytest.mark.MARKER_NAME.__call__`
```

Comment:
This uses rst syntax, so the monospace references should use double backticks:

``` 
... demonstrating the usage of ``pytest.mark.MARKER_NAME.with_args`` in ...
```

If you could fix that after rebasing that would be great. 😁